### ISSUE TITLE
Update to Biolink v3.5.0

### ISSUE THREAD

**Issue by ecwood** [RTXteam/RTX-KG2#302]
Based on https://github.com/RTXteam/RTX-KG2/issues/301#issuecomment-1610336171, we need to update to Biolink v3.5.0. Currently, we are on v3.1.2. Since we are so out of date, I am creating this issue to document some past changes that might be sticky for us.

For example, in [Biolink v3.2.7](https://github.com/biolink/biolink-model/releases/tag/v3.2.7), [a lot changed with KEGG IRIs](https://github.com/biolink/biolink-model/pull/1271/commits/7c1046ddb8f4b58e084b8875b3dd9b5f46205f22).

Unfortunately, Biolink v3.5.0 doesn't exist yet to start working with.

**Comment by ecwood** [comment]
From @saramsey:

> Oh yes, so on a related note, but motivated by a different issue, we will need to start developing a file of Biolink predicates that have been deprecated since the Biolink version that xDTD is currently based on (will need to ask Chunyu), and to curate which Biolink predicates (and qualifiers, if necessary) those map to. That will require some creativitiy.

**Comment by ecwood** [comment]
[v3.2.0](https://github.com/biolink/biolink-model/releases/tag/v3.2.0): I don't see anything we have to change.

[v3.2.1](https://github.com/biolink/biolink-model/releases/tag/v3.2.1): There's a change with KGX, but I don't think this impacts us.

[v3.2.2](https://github.com/biolink/biolink-model/releases/tag/v3.2.2):
- UniProt PURLs

[v3.2.3](https://github.com/biolink/biolink-model/releases/tag/v3.2.3): This doesn't impact us.

[v3.2.4](https://github.com/biolink/biolink-model/releases/tag/v3.2.4): This shouldn't matter to us (`slot_usage` information about `retrieval source`). Should we be using `retrieval source` as the category for our upstream sources, rather than `information content entity`? What is the difference between this and `information resource`?

[v3.2.5](https://github.com/biolink/biolink-model/releases/tag/v3.2.5): This doesn't impact us.

[v3.2.6](https://github.com/biolink/biolink-model/releases/tag/v3.2.6):

**Comment by saramsey** [comment]
@ecwood any thoughts about the 3.3.X and 3.4.X releases?

**Comment by ecwood** [comment]
> @ecwood Thank you! Any thoughts about the 3.3.X and 3.4.X releases?

Yes! I updated the comment with information about the other releases.

**Comment by saramsey** [comment]
Thanks so much. I used the `master` branch of `biolink-model.yaml` to guide the changes that I made to `predicate-remap.yaml` for #305. Hopefully that helps move us toward 3.5.0 compliance.

**Comment by ecwood** [comment]
I expected that we'd need to change the UniProt URL. However, `run-validation-tests.sh` on the master branch is not failing.

@saramsey Do you know why it (specifically `validate_curies_to_urls_map_yaml.py`) wouldn't fail despite these lines being different?

https://github.com/RTXteam/RTX-KG2/blob/9aa895416f9ff7abbae5fb699869244ea908a2c0/curies-to-urls-map.yaml#L716-L717

https://github.com/biolink/biolink-model/blob/5b9c2834e6ae548f65f1819fb09e390e7aa3f307/biolink-model.yaml#L143:

```
  UniProtKB: 'http://purl.uniprot.org/uniprot/'
```

**Comment by acevedol** [comment]
Strange that the validation tests don't mind the differences in the URLs. I don't know why that would be. 
Just to remind myself because I've been out so much, in addition to the changes above to catch up to the current biolink model, we are also addressing issue #281, adding a domain_range_exclusion boolean type property to edges. 

@ecwood , do you have a preference on where I get started on the changes?

**Comment by ecwood** [comment]
@acevedol Some of the changes have already been made (#306), and all of that has been done in `kg2.8.4-prep`, so let's stick with that.

> we are also addressing issue https://github.com/RTXteam/RTX-KG2/issues/281, adding a domain_range_exclusion boolean type property to edges

Yes, but unfortunately, we don't have the schema for how we will get this data yet, so it's not much use to start the code for it yet.

**Comment by ecwood** [comment]
While testing out bc079cf (to ensure the validation scripts are using the correct version of biolink), we got this error:

```
Reading ontology JSON file: /home/ubuntu/kg2-build/biolink-model.owl.json; size: 2213.82 KiB
Traceback (most recent call last):
  File "/home/ubuntu/kg2-code/validate_kg2_util_curies_urls_categories.py", line 62, in <module>
    assert category_curie in biolink_categories_ontology_depths, category_curie
AssertionError: biolink:InformationResource
```

**Comment by ecwood** [comment]
With [8a63bbc](https://github.com/RTXteam/RTX-KG2/commit/8a63bbc2110cab025165afdb7e722c78a8da6862), the validator is not failing anymore.

**Comment by ecwood** [comment]
Per @saramsey, `validate_curies_to_urls_map_yaml.py` is validating CURIEs only, not the URLs. In an enhancement, we should have the script compare the URLs to what is in Biolink.

**Comment by ecwood** [comment]
Now that [#319](https://github.com/RTXteam/RTX-KG2/issues/319) and [#320](https://github.com/RTXteam/RTX-KG2/issues/320) have been finished, tested, and are passing, do we need to do anything else to verify that we are Biolink 3.5.0 compliant? Do we need to do [#306](https://github.com/RTXteam/RTX-KG2/issues/306) to verify compliance?

**Comment by saramsey** [comment]
Yes, closing.
